fuel/datasets/hdf5.py

```diff
@@ -227,6 +228,34 @@ def parse_split_array(split_array):
                 split_dict[split][source] = (start, stop, comment)
         return split_dict
 
+    @staticmethod
+    def unordered_fancy_index(request, indexable):
+        """Safe list indexing.
+
+        Some objects, such as h5py datasets, only support list indexing
+        if the list is ordered.
```

Comment:
I am not 100% sure, but I think that one should say "sorted list" not "ordered list". Because all list are ordered, `OrderedDict` is ordered, but only sorted lists contain their elements in ascending order.